RELEASING.md

```diff
@@ -0,0 +1,21 @@
+# Releasing
+
+Builds and releases are automated with GitHub Actions and GoReleaser. There are a few manual steps to complete:
+
+1. Start the release action: 
+   
+  ```shell
+  git checkout master
+  git tag vX.Y.Z
+  git push upstream tag vX.Y.Z
+  ```
+
+  Where `upstream` is the remote name pointing to the canonical/terraform-provider-maas repository. 
+  
+  The provider versions follow [semantic versioning](https://semver.org/), and the release action can be viewed under [Actions](https://github.com/canonical/terraform-provider-maas/actions).
+
+2. Publish the release: 
+   
+   The action creates a "draft" release. Go to [Releases](https://github.com/canonical/terraform-provider-maas/releases) to open it, select `edit`, and select `Publish release` if you are happy. 
+
+3. Verify the published release looks good by reviewing [Releases](https://github.com/canonical/terraform-provider-maas/releases).
```

Comment:
We might also instruct the future provider maintainers to confirm that the provider page in the HashiCorp registry displays the released version vX.Y.Z as the latest. This can take some time and based on my experience can last couple of minutes up to half an hour.

For example, https://registry.terraform.io/providers/canonical/maas/latest should point to vX.Y.Z or https://registry.terraform.io/providers/canonical/maas/X.Y.Z is a valid link.